src/test/java/net/snowflake/client/authentication/IdTokenIT.java

```diff
@@ -0,0 +1,85 @@
+package net.snowflake.client.authentication;
+
+import static net.snowflake.client.authentication.AuthConnectionParameters.getStoreIDTokenConnectionParameters;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.io.IOException;
+import net.snowflake.client.category.TestTags;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@Tag(TestTags.AUTHENTICATION)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```

Comment:
IMO, it's more readable to have three tests that test one thing run in order than one long test, but it can be changed  to one test that will test the whole flow